src/masternodes/rpc_accounts.cpp

```diff
@@ -1184,6 +1192,9 @@ UniValue listaccounthistory(const JSONRPCRequest& request) {
                 return txs.count(pwtx->GetHash()) || startBlock > index->nHeight || index->nHeight > maxBlockHeight;
             },
             [&](COutputEntry const & entry, CBlockIndex const * index, CWalletTx const * pwtx) {
+                if(txn != std::numeric_limits<uint32_t>::max() && txn != entry.vout) {
+                    return true;
+                }
```

Comment:
do we need this? still feels like a `filteration`

Reply:
Good point, It should be better to be
```
if (txn != std::numeric_limits<uint32_t>::max() && index->nHeight == maxBlockHeight && txn < entry.vout)
```